### PR TITLE
Add org.fooyin.fooyin

### PR DESCRIPTION
**Pull request by eclipseo**
Fooyin

Fooyin is a music player built around customisation. It offers a growing list of widgets to manage and play your local music collection. It's extendable through the use of plugins and scriptable using FooScript.

Features:
- Customisable layout
- Gapless playback
- Filter and search music collection
- Create and manage playlists
- Extendable through plugins
- Tag editing

### Please confirm your submission meets all the criteria

- [x] Please describe your application briefly.
- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link: https://github.com/ludouzi/fooyin/issues/99** **In progress, ownership shall be his too**
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed. **Software author has acquired the domain name.**
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* **PR were sent and merged.**

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

**Software author must have co-ownership of the repo if approved: https://github.com/ludouzi/**

_This is my first Flatpak, any advice is greatly appreciated._

Linting:

```
flatpak run --command=flatpak-builder-lint org.flatpak.Builder manifest org.fooyin.fooyin.yaml
no error

flatpak run --command=flatpak-builder-lint org.flatpak.Builder appstream org.fooyin.fooyin.metainfo.xml
? Validation was successful.

flatpak run --command=flatpak-builder-lint org.flatpak.Builder repo repo
{
    "errors": [
        "appstream-remote-icon-not-mirrored",
        "appstream-external-screenshot-url"
    ],
    "message": "Please consult the documentation at https://docs.flathub.org/docs/for-app-authors/linter"
}

flatpak run org.fooyin.fooyin
Detected locale "C" with character encoding "ANSI_X3.4-1968", which is not UTF-8.
Qt depends on a UTF-8 locale, and has switched to "C.UTF-8" instead.
If this causes problems, reconfigure your locale. See the locale(1) manual
for more information.
Loaded "qt" translations for locale "en_US" from "/usr/translations"
Failed to load "fooyin" translations for locale "en_US" from "/app/share/fooyin/translations"
Detected locale "C" with character encoding "ANSI_X3.4-1968", which is not UTF-8.
Qt depends on a UTF-8 locale, and has switched to "C.UTF-8" instead.
If this causes problems, reconfigure your locale. See the locale(1) manual
for more information.
Loaded "qt" translations for locale "zh_CN" from "/usr/translations"
Loaded "fooyin" translations for locale "zh_CN" from "/app/share/fooyin/translations"
```

Translations do work though.

Not sure how to handle the mirroring yet.

### CODE REVIEW

File: flathub.json
Comment (hfiguiere) [inline flathub.json:1]: remove this file. you don't need it.

File: org.fooyin.fooyin.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+app-id: org.fooyin.fooyin
```
Comment (hfiguiere) [inline org.fooyin.fooyin.yaml:1]: `fooyin.org` doesn't work. (http or https)

File: org.fooyin.fooyin.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        commit: ebf4c5bbb1ce85a71764122cbc3cc3f556e66586
+        x-checker-data:
+          type: anitya
+          project-id: 1982
+          stable-only: true
+  - name: fooyin
+    buildsystem: cmake-ninja
+    builddir: true
+    config-opts:
+      - '-DBUILD_ALSA=OFF'
```
Comment (hfiguiere) [inline org.fooyin.fooyin.yaml:75]: why disabling ALSA?

File: org.fooyin.fooyin.desktop
Comment (hfiguiere) [inline org.fooyin.fooyin.desktop:1]: this should be upstream.

File: org.fooyin.fooyin.metainfo.xml
Comment (hfiguiere) [inline org.fooyin.fooyin.metainfo.xml:1]: this should be upstream.

Comment (hfiguiere) [reply org.fooyin.fooyin.metainfo.xml:1]: also it's never referenced

File: org.fooyin.fooyin.desktop
Comment (hfiguiere) [reply org.fooyin.fooyin.desktop:1]: also it's never referenced

**Comment by hfiguiere** [comment]
> Not sure how to handle the mirroring yet.

Just remove "remote" icons from the appstream file.

**Comment by hfiguiere** [comment]
bot, build org.fooyin.fooyin

**Comment by hfiguiere** [comment]
> > Not sure how to handle the mirroring yet.
> 
> Just remove "remote" icons from the appstream file.

and I don't see any. Let's see what buildbot has to say.

File: org.fooyin.fooyin.metainfo.xml
Comment (eclipseo) [reply org.fooyin.fooyin.metainfo.xml:1]: Oh this is upstream, I must have forgot to remove them.

File: org.fooyin.fooyin.yaml (hunk 1 above)
Comment (eclipseo) [reply org.fooyin.fooyin.yaml:1]: Not yet. The author just acquired the domain:

```
Domain Name: fooyin.org
Registry Domain ID: f2f11bb40bb9429d923ba43e6c871550-LROR
Registrar WHOIS Server: http://whois.porkbun.com
Registrar URL: https://porkbun.com
Updated Date: 2024-05-18T08:53:02Z
Creation Date: 2024-05-18T08:53:01Z
```

File: org.fooyin.fooyin.desktop
Comment (eclipseo) [reply org.fooyin.fooyin.desktop:1]: Yes this was remnant before the PR were merge upstream.

File: org.fooyin.fooyin.yaml (hunk 2 above)
Comment (eclipseo) [reply org.fooyin.fooyin.yaml]: Ok found the issue, it is reenabled.

**Comment by eclipseo** [comment]
bot, build org.fooyin.fooyin

**Comment by eclipseo** [comment]
Ok, I've tested locally, it works fine.

As said, I don't have any remote icon, so I don't get: appstream-remote-icon-not-mirrored.

I do have an external screenshot, but from what I've understood --mirror-screenshots-url=URL will handle it, right?

File: org.fooyin.fooyin.yaml (hunk 1 above)
Comment (bbhtt) [reply org.fooyin.fooyin.yaml:1]: Works now 

image

File: flathub.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+    "automerge-flathubbot-prs": false,
+    "disable-external-data-checker": false,
+    "require-important-update": false
+}
```
Comment (bbhtt) [inline flathub.json:5]:
```suggestion
{
    "disable-external-data-checker": false,
    "require-important-update": false
}
```

Please don't use automerge we will block it soon

File: org.fooyin.fooyin.yaml [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  - '--filesystem=/run/media'
+  - '--own-name=org.mpris.MediaPlayer2.fooyin'
+  - '--share=ipc'
+  - '--share=network'
+  - '--socket=fallback-x11'
+  - '--socket=pulseaudio'
+  - '--socket=wayland'
+  - '--system-talk-name=org.freedesktop.UDisks2'
+  - '--talk-name=org.freedesktop.Notifications'
+  - '--talk-name=org.gtk.Notifications'
```
Comment (bbhtt) [inline org.fooyin.fooyin.yaml:25]: Are these two required? They don't go through the portal? 

Also why `org.gtk.Notifications`? It seems to be a Qt app.

File: org.fooyin.fooyin.yaml [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    sources:
+      - type: git
+        url: https://github.com/KDAB/KDSingleApplication
+        tag: 'v1.1.0'
+        commit: cb0c664b40d3b31bad30aa3521eff603162ed0bd
+        x-checker-data:
+          type: anitya
+          project-id: 370439
+          stable-only: true
+  - name: taglib
```
Comment (bbhtt) [inline org.fooyin.fooyin.yaml:73]: It's in the runtime https://invent.kde.org/packaging/flatpak-kde-runtime/-/blob/4930296f1a6d17eee3863a03c554780bb94baf62/org.kde.Sdk.json.in#L1317

File: org.fooyin.fooyin.yaml [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+  - '--share=ipc'
+  - '--share=network'
+  - '--socket=fallback-x11'
+  - '--socket=pulseaudio'
+  - '--socket=wayland'
+  - '--system-talk-name=org.freedesktop.UDisks2'
+  - '--talk-name=org.freedesktop.Notifications'
+  - '--talk-name=org.gtk.Notifications'
+  - '--talk-name=org.kde.StatusNotifierWatcher'
+  - '--talk-name=org.mpris.MediaPlayer2.*'
```
Comment (bbhtt) [inline org.fooyin.fooyin.yaml:27]: Why is this necessary?

File: org.fooyin.fooyin.yaml [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+  - '--own-name=org.mpris.MediaPlayer2.fooyin'
+  - '--share=ipc'
+  - '--share=network'
+  - '--socket=fallback-x11'
+  - '--socket=pulseaudio'
+  - '--socket=wayland'
+  - '--system-talk-name=org.freedesktop.UDisks2'
+  - '--talk-name=org.freedesktop.Notifications'
+  - '--talk-name=org.gtk.Notifications'
+  - '--talk-name=org.kde.StatusNotifierWatcher'
```
Comment (bbhtt) [inline org.fooyin.fooyin.yaml:26]: I don't see it having tray functionality?

**Comment by bbhtt** [comment]
I'm seeing a blank window from second launch


https://github.com/flathub/flathub/assets/62639087/9c1ad71b-8ae3-4d44-b4c9-ff37b457ceee

File: org.fooyin.fooyin.yaml (hunk 4 above)
Comment (bbhtt) [reply org.fooyin.fooyin.yaml:25]: I don't see it having the ability to send any desktop notifications from a first look.

File: flathub.json
Comment (bbhtt) [reply flathub.json:1]: This is not resolved. automerge will be blocked, the rest are the default.

File: org.fooyin.fooyin.yaml (hunk 4 above)
Comment (eclipseo) [reply org.fooyin.fooyin.yaml]: Yes indeed, I thought this was needed for MPris notifications

File: org.fooyin.fooyin.yaml (hunk 5 above)
Comment (eclipseo) [reply org.fooyin.fooyin.yaml]: Surprising, thanks.

File: org.fooyin.fooyin.yaml (hunk 7 above)
Comment (eclipseo) [reply org.fooyin.fooyin.yaml]: >  '--system-talk-name=org.freedesktop.UDisks2'

is needed, otherwise it throws an error at startup regarding udisk.

Removing notification stuff.

**Comment by eclipseo** [comment]
bot, build org.fooyin.fooyin

**Comment by eclipseo** [comment]
> I'm seeing a blank window from second launch
> Screencast.mp4

I can't reproduce this on Plasma 6. I will need to try with a GNOME system like yours maybe.

File: org.fooyin.fooyin.yaml (hunk 6 above)
Comment (eclipseo) [reply org.fooyin.fooyin.yaml]: I thought this was needed with own-name based on other YAML I have seen.

**Comment by eclipseo** [comment]
Screenshot_20240519_062555


In the screemcast, on your second run, we see the icon of Fooyin in your taskbar.

File: org.fooyin.fooyin.yaml (hunk 7 above)
Comment (bbhtt) [reply org.fooyin.fooyin.yaml]: I'm talking about `org.kde.StatusNotifierWatcher`

**Comment by bbhtt** [comment]
> In the screencast, on your second run, we see the icon of Fooyin in your taskbar.

But the window is invisible which is the issue.

File: org.fooyin.fooyin.yaml (hunk 7 above)
Comment (bbhtt) [reply org.fooyin.fooyin.yaml]: I see it was removed in https://github.com/flathub/flathub/pull/5261/commits/e767076c77290bbe179cb7ce98e2b7837d64dcb5 resolving.

File: org.fooyin.fooyin.yaml (hunk 6 above)
Comment (bbhtt) [reply org.fooyin.fooyin.yaml]: For the MPRIS notification from the app you only need `--own-name=org.mpris.MediaPlayer2.$appname` if the app uses the app-id there (as it should) you don't even need anything as Flatpak by default allows access to `--own-name=org.mpris.MediaPlayer2.$FLATPAK_ID`

Unless the app is a client the talk name and full control over all interfaces isn't required.

Comment (bbhtt) [reply org.fooyin.fooyin.yaml]: Seems to be removed by https://github.com/flathub/flathub/pull/5261/commits/e767076c77290bbe179cb7ce98e2b7837d64dcb5 resolving.

File: flathub.json
Comment (bbhtt) [reply flathub.json:1]: @eclipseo please resolve this one.

**Comment by ludouzi** [comment]
I've just tried the latest test build in Gnome 46.1 and it runs fine on subsequent launches, so I'm not sure what's happening in your case. I'll have to do some testing.

**Comment by eclipseo** [comment]
> > In the screencast, on your second run, we see the icon of Fooyin in your taskbar.
> 
> But the window is invisible which is the issue. I can only get it back on second launch by rm-ing all app data from `~/.var/app/app-id` and launching again so that it is a "fresh" start.

What os/GNOME are you using so I can try to repro?

**Comment by bbhtt** [comment]
It's Fedora 39/GNOME 45 inside GNOME Boxes.

**Comment by bbhtt** [comment]
Looks like I can't repro on my host running F40

This comment needs to be solved https://github.com/flathub/flathub/pull/5261#discussion_r1605804782

File: flathub.json
Comment (eclipseo) [reply flathub.json:1]: Wait, remove the entire flathub.json? Don't need this to configure update rules?

Comment (eclipseo) [reply flathub.json:1]: oh ok.

Comment (bbhtt) [reply flathub.json:1]: > Wait, remove the entire flathub.json? Don't need this to configure update rules?

No, all you need is the x-checker para for a module in the manifest

**Comment by eclipseo** [comment]
Thank you.
